Add shared database docs

rmweir opened the pull request:
Preview platform: https://deploy-preview-314--vcluster-docs-site.netlify.app/docs/platform/administer/connector/database
Preview vcluster: https://deploy-preview-314--vcluster-docs-site.netlify.app/docs/vcluster/next/configure/vcluster-yaml/control-plane/components/backing-store/database/external

Piotr1215 commented on platform/administer/shared-database/shared-database.mdx:
```diff
@@ -0,0 +1,82 @@
+---
+title: Shared database vCluster Platform
+sidebar_label: Shared Database
+sidebar_position: 1
```
Could you also add `description` to the front matter. This helps discover the docs by search engines.

```suggestion
sidebar_position: 1
description: Learn how to use shared database with vcluster.
```

rmweir replied:
Done, I took the description you provided :P.

Piotr1215 commented on platform/administer/shared-database/shared-database.mdx:
```diff
@@ -0,0 +1,82 @@
+sidebar_position: 1
+---
+
+The shared database feature allows an admin to configure a database server to be used to manage backing stores for multiple vCluster. For each vCluster, the feature automatically creates individual databases and
```
```suggestion
The shared database feature allows an admin to configure a database server to be used to manage backing stores for multiple virtual clusters. For each virtual cluster, the feature automatically creates individual databases and
```

Piotr1215 commented on platform/administer/shared-database/shared-database.mdx:
````diff
@@ -0,0 +1,82 @@
+    * user
+
+The template for a shared database connector secret is:
+```
````
We would like to have titles on the code blocks, maybe something like:
\```yaml title="Shared database secret"

Piotr1215 commented on platform/administer/shared-database/shared-database.mdx:
````diff
@@ -0,0 +1,82 @@
+```
+
+1. Replace all values surrounded by angular brackets with your own value and save to a file as `<filename>`.
+2. Create the secret:
+`kubectl create -f <filename>`
````
```suggestion
2. Create the secret: `kubectl create -f <filename>`
```

Piotr1215 commented on platform/administer/shared-database/shared-database.mdx:
````diff
@@ -0,0 +1,82 @@
+with a ClusterIP service named `mysql` in the `vcluster-platform` namespace (let us presume this is the namespace vCluster Platform is installed in). This could
+instead be any other endpoint, such as an AWS RDS MySQL endpoint.
+
+```
````
Similar here, maybe:
\```yaml title="Endpoint secret"

rmweir replied:
Used different title but done.

Piotr1215 reviewed (commented):
Thank you for submitting the doc @rmweir , added a few comments/suggestions.

rmweir commented:
@Piotr1215 addressed feedback and added a [warning](https://github.com/loft-sh/vcluster-docs/pull/314/files#diff-6885da81aabd0fa338bc2d6fc9553d177cb9649272e49de9bb79d8a6b02c55e2R61) regarding the database secrets.

sowmyav27 commented on platform/administer/shared-database/shared-database.mdx:
```diff
@@ -0,0 +1,85 @@
+1. Replace all values surrounded by angular brackets with your own value and save to a file as `<filename>`.
+2. Create the secret: `kubectl create -f <filename>`
+
+That's it. A new vCluster can now reference the created connector secret by setting the `controlPlane.backingStore.database.external.connector` field in the vCluster config
```
Can we add a vCluster.yaml here instead so it will be easier for users to configure ?

rmweir replied:
Added a full separate doc for vCuster now.

Piotr1215 reviewed (commented):
Thank you for writing the doc @rmweir. Looks like there are some unresolved links issues in the build. Could you please [allow edits by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork), so I can help you with it?

Piotr1215 reviewed (commented):
@rmweir could you please check if my style and writing modifications didn't mess up with the content?

rmweir commented on platform/administer/connector/database.mdx:
```diff
@@ -2,37 +2,42 @@
 
+import InstallNextSteps from '../../_partials/install/install-next-steps.mdx';
+
+The database connector feature allows an admin to configure a database server to manage backing stores for multiple virtual clusters. For each virtual cluster, the feature automatically creates individual databases and non-privileged users. Read more about the use of external databases in the [External Database documentation](/docs/vcluster/configure/vcluster-yaml/control-plane/components/backing ...
```
`the feature automatically creates individual databases and non-privileged users` should be `the feature automatically creates an independent database and non-privileged user`. Each virtual cluster only gets one of these.
